onnxruntime/core/session/environment.cc

```diff
@@ -106,7 +106,10 @@ Internal copy node
 }
 
 Environment::~Environment() {
-  ::google::protobuf::ShutdownProtobufLibrary();
+  // TODO:  we need to support 2 things:
+  // we need to NOT shutdown the lib so that folks can create more Env's and use them
+  // we need to have a way to NOT leak memory since ORT can be a DLL
+  //::google::protobuf::ShutdownProtobufLibrary();
```

Comment:
are we going to RI this? or does this need to be addressed before we bring this back inbox?

Reply:
I think this should be addressed before we bring this back into inbox. There's an email thread regarding this and bug(s) will fall out of it.

---
In reply to: [354568749](https://github.com/microsoft/onnxruntime/pull/2562#discussion_r354568749) [](ancestors = 354568749)

Reply:
I suggest we move "::google::protobuf::ShutdownProtobufLibrary()" to Dllmain, and avoid calling it here. 
If the application is statically linked to onnxruntime, then it can shutdown protobuf by himself (if it's really needed).
Otherwise, we could do it in dllmain.

Reply:
Love both of these options.   We are going to chat with Pranav i think later today as well to figure out how ORT overall wants to work with shutting down protobuf , static link customers, and C_ABI customers both.  thanks changming !

---
In reply to: [354578010](https://github.com/microsoft/onnxruntime/pull/2562#discussion_r354578010) [](ancestors = 354578010)